app/controllers/works_controller.rb

```diff
@@ -0,0 +1,89 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @works = Work.all.top(25)
+    @movies = []
+    @books = []
+    @albums = []
+    @works.each do |work|
+      if work.category == "movie"
+        @movies << work
+      elsif work.category == "book"
+        @books << work
+      else
+        @albums << work
+      end
+    end
+  end
+
+  def main
+    @works = Work.all.top(25)
+    @movies = []
+    @books = []
+    @albums = []
+    @works.each do |work|
+      if work.category == "movie"
+        @movies << work
+      elsif work.category == "book"
+        @books << work
+      else
+        @albums << work
+      end
+    end
+  end
+
+  def show;  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = 'Work added!'
+      redirect_to work_path(@work.id)
+    else
+      flash.now[:error] = 'Work not added'
```

Comment:
A more specific set of validation errors would be better, use `@work.errors.messages`.